util/chunk/list.go

```diff
@@ -145,6 +155,15 @@ func (l *List) Reset() {
 	l.consumedIdx = -1
 }
 
+// Clear frees all the allocated chunks
+func (l *List) Clear() {
+	l.memTracker.Consume(-l.memTracker.BytesConsumed())
+	l.freelist = l.freelist[:0]
+	l.chunks = l.chunks[:0]
```

Comment:
Slice cutting does not release the reference of underlying array, so it does not free the memory actually.